ml/src/config.py

```
@@ -183,6 +198,37 @@ def _validate(cfg: dict) -> None:
+    if not isinstance(stride, (int, float)) or isinstance(stride, bool):
+        raise ValueError("preprocessing.patch_stride_fraction must be a number")
+    if not 0.0 < stride <= 1.0:
+        raise ValueError(
+            "preprocessing.patch_stride_fraction must be in (0, 1], got "
+            f"{stride}: 1.0 is a non-overlapping grid and anything above it "
+            "would leave gaps of soil the model never sees"
+        )
+
+    minimum = pre["min_patches"]
+    if not isinstance(minimum, int) or isinstance(minimum, bool):
+        raise ValueError("preprocessing.min_patches must be an integer")
```

Ruff (0.16.3)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)